Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -108,6 +131,48 @@ public void Serialize(IIonWriter writer, object item)
             writer.StepOut();
         }
 
+        private object InvokeIonConstructor(ConstructorInfo ionConstructor, IIonReader reader)
+        {
+            var parameters = ionConstructor.GetParameters();
+            var arguments = new object[parameters.Length];
+
+            if (parameters.Length > 0)
+            {
+                // Compute mapping between parameter names and index in parameter array.
+                var paramIndexMap = new Dictionary<string, int>();
+                for (int i = 0; i < parameters.Length; i++)
+                {
+                    var param = (IonPropertyName)parameters[i].GetCustomAttribute(typeof(IonPropertyName));
+                    if (param == null)
+                    {
+                        throw new NotSupportedException(
+                            $"Parameter '{parameters[i].Name}' is not specified with the [IonPropertyName] attribute " +
+                            $"for {targetType.Name}'s IonConstructor");
+                    }
+                    
+                    paramIndexMap.Add(param.Name, i);
+                }
+                
+                reader.StepIn();
+
+                // Iterate through reader to determine argument values to pass into constructor.
+                IonType ionType;
+                while ((ionType = reader.MoveNext()) != IonType.None)
```

Comment:
What happens if I have a property set within this Ion struct? They will be ignored and skipped. I think we want to include them.

Reply:
I'm unclear what you're asking.
This code is simply searching the Ion for the `IonConstructor`s arguments so we can invoke the constructor.

Reply:
I believe Ian is referring to is that properties not included in the constructor should also be set. @justing-bq please confirm behaviour.
@HomayounA fyi.

Reply:
@kylepbit Ian and I discussed that previously and we agreed to just invoke the constructor and leave the other properties as is. 
@alpian Did you change your mind about that?